### PR TITLE
Kairos downsampling: better fix, works with multi-source charts

### PR DESCRIPTION
**Pull request by zachm**
This is a followup to pull/2898 and #2894. We should not enforce a 10s avg aggregation if the downsampling is undefined. For example, if you want to do a count on a multi data source chart, this breaks it entirely.

Instead, we add a check that, if the downsampling aggregation name is undefined, we do nothing. i.e., the default empty select box behaves just like selecting (NONE) now.

### CODE REVIEW

**Comment by torkelo** [comment]
why not enforce a default? you can change the downsample to count if you want. Sounds like a very bad idea to encourage no downsampling as it can easily crash your browser that way buy having kariosdb return millions of data points.

**Comment by zachm** [comment]
@torkelo, my impression is that downsampling only works as a feature if you use a single data source - otherwise the box disappears. Is downsampling supposed to work for multi data source charts?

I'm very happy to have a default in the UI downsampling box. (Previously, the default was blank and therefore broken.) But this doesn't solve downsampling for multi-data source charts... what do you think makes sense there?

Basically I don't care what the default is as long as it's changeable and not inherently broken.

**Comment by torkelo** [comment]
hm.. the kairosdb data source looks a little hacky when it comes to this, the down sampling function should be on the each query, looks like it needs some refactoring or rethink

**Comment by bergquist** [comment]
Kairosdb datasource is moved from the grafana repo into a plugin repo https://github.com/Kratos-ISE/grafana-kairosdb-datasource-plugin

**Comment by torkelo** [comment]
that is not the repo url, it's https://github.com/grafana/datasource-plugin-kairosdb
